addon/record-data.js

```diff
@@ -207,7 +217,14 @@ export default class M3RecordData {
       this.id = jsonApiResource.id + '';
     }
 
-    if (this._notifyProjectionProperties(changedKeys)) {
+    // by default, always notify projections when we receive data.  We might
+    // not have been asked to calculate changes if the base record data has
+    // no record, but we might still have records instantiated for our
+    // projections.
+    //
+    // Notifications are explicitly suppressed when we're using `pushData`
+    // synthetically while resolving nested record data
+    if (!suppressNotifications && this._notifyProjectionProperties(changedKeys)) {
```

Comment:
Is `suppressNotifications` actually `suppressProjectionNotification`? Otherwise, we dont't honor it on `232`
We seem to be increasing the number of notification flags, wondering if there is a way to refactor away?

Reply:
good call; i'll rename the flag

re: flags I very much agree; i'd like to see if we can clean this up in a followup pr